pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -139,14 +141,26 @@ def transfer_distrib_spawn_state_on_fit_end(self, results):
             # TODO: is there a better way than accessing trainer through model -> trainer?
             if not self.lightning_module.trainer.testing and best_model_path is not None and len(best_model_path) > 0:
                 last_path = re.sub(".ckpt", ".tmp_end.ckpt", best_model_path)
-                xm.save(self.lightning_module.state_dict(), last_path)
+                self.try_save(self.lightning_module.state_dict(), last_path)
 
             if self.global_rank == 0:
                 # todo, pass complete checkpoint as state dictionary
                 self.mp_queue.put(best_model_path)
                 self.mp_queue.put(last_path)
                 self.mp_queue.put(results)
 
+    def try_save(self, state_dict: Dict, path: str) -> None:
+        """
+        Saving with xm.save can failed to meet rendez-vous.
+        Therefore, we will try several times to do so.
+        """
+        for _ in range(self._repeat_save_on_fail):
+            try:
+                xm.save(state_dict, path)
+                break
+            except RuntimeError:
+                sleep(0.001)
```

Comment:
how about the last try, there is no final error if all fails?

Reply:
I moved it into a try / catch instead.
I still don't know why it is so unstable, but it shouldn't affect training.